libglusterfs/src/mem-pool.c

```diff
@@ -91,35 +91,28 @@ gf_mem_set_acct_info(struct mem_acct *mem_acct, struct mem_header *header,
                      size_t size, uint32_t type, const char *typestr)
 {
     struct mem_acct_rec *rec = NULL;
-    bool new_ref = false;
+    uint64_t num_allocs;
 
     if (mem_acct != NULL) {
         GF_ASSERT(type <= mem_acct->num_types);
 
         rec = &mem_acct->rec[type];
+        if (!rec->typestr) {
+            rec->typestr = typestr;
+        }
+        num_allocs = GF_ATOMIC_INC(rec->num_allocs);
+        rec->max_size = max(rec->max_size, (num_allocs * size));
+        rec->max_num_allocs = max(rec->max_num_allocs, num_allocs);
```

Comment:
I think we don't need this. Otherwise this would require an atomic to update each one in a safe way.

Reply:
done, move both updates if debug is enabled